tests/qiskit_convert_test.py

```diff
@@ -1285,14 +1287,47 @@ def test_ifelseop_multi_bit_cond() -> None:
     circuit.h(q1)
     circuit.measure(q0, c0)
     circuit.measure(q1, c1)
-    with circuit.if_test((creg, 2)):
+    # Condition is on a register not a bit
+    with circuit.if_test((creg, 2)) as else_:
         circuit.x(q0)
         circuit.x(q1)
+    with else_:
+        circuit.y(q0)
+        circuit.y(q1)
     circuit.measure(q0, c0)
     circuit.measure(q1, c1)
-    # This currently gives an error as register exp not supported.
-    with pytest.raises(NotImplementedError):
-        qiskit_to_tk(circuit)
+    tkc: Circuit = qiskit_to_tk(circuit)
+    tkc.name = "test_circ"
+
+    expected_circ = Circuit()
+    expected_circ.name = "test_circ"
+    qreg_tk = expected_circ.add_q_register("q", 2)
+    creg_tk = expected_circ.add_c_register("c", 2)
+    expected_circ.H(qreg_tk[0])
+    expected_circ.H(qreg_tk[1])
+    expected_circ.Measure(qreg_tk[0], creg_tk[0])
+    expected_circ.Measure(qreg_tk[1], creg_tk[1])
+
+    x_circ2 = Circuit()
+    x_circ2.name = "If"
+    x_qreg = x_circ2.add_q_register("q", 2)
+    x_circ2.X(x_qreg[0]).X(x_qreg[1])
+    expected_circ.add_circbox(
+        CircBox(x_circ2), [qreg_tk[0], qreg_tk[1]], condition=reg_eq(creg_tk, 2)
+    )
+
+    y_circ2 = Circuit()
+    y_circ2.name = "Else"
+    y_qreg = y_circ2.add_q_register("q", 2)
+    y_circ2.Y(y_qreg[0]).Y(y_qreg[1])
+    expected_circ.add_circbox(
+        CircBox(y_circ2), [qreg_tk[0], qreg_tk[1]], condition=reg_neq(creg_tk, 2)
+    )
+
+    expected_circ.Measure(qreg_tk[0], creg_tk[0])
+    expected_circ.Measure(qreg_tk[1], creg_tk[1])
+
+    assert expected_circ == tkc
 
```

Comment:
Can we add one more test where there is no `else` condition to make sure that works?

Reply:
good idea. A test for a single branch using more qubits/bits and a different circuit.